src/demo.re

```diff
@@ -0,0 +1,66 @@
+/* 
+   Goal: 
+     conjugate('estudiar', { tense: 'present', subject: 'ella' });
+       should return estudia
+ */
+
+type subject = 
+| Yo
+| Tu
+| Ella
+| Nosotras
+| Vosotras
+| Ellas;
+
+type ending = 
+| Ar
+| Er
+| Ir;
+
+let getSubjectVariant subject => {
```

Comment:
Typically in Reason you would group together a type and its main operations into a single module. E.g.:

```reason
module Subject = {
  /* The type name becomes just `t` to avoid repetition when accessed from outside the module */
  type t =
    | Yo
    | Tu
    | Ella
    | Nosotras
    | Vosotras
    | Ellas;
  let ofString =
    fun
    | "yo" => Yo
    | "tu" => Tu
    | "ella" => Ella
    | "nosotras" => Nosotras
    | "vosotras" => Vosotras
    | _ => Ellas;
};
```

... and similarly for the `ending` type. Then you'd be able to use them like `Subject.ofString`, `Ending.ofString`. It makes the API uniform.